src/Components/NotificationDropdown.js

```diff
@@ -0,0 +1,108 @@
+import React, { Fragment, Component } from "react";
+import { withRouter } from 'react-router';
+import { MDBDropdown, MDBDropdownToggle, MDBDropdownMenu, MDBDropdownItem } from "mdbreact";
+import axios from "axios";
+import './styles/NotificationDropdown.css'
+
+function NotifIcon(props) {
+    const notiftype = props.value._notification_type;
+    const isread = props.value.is_read ? 'var(--medgray)' : 'var(--saturatedpurple)';
+    switch(notiftype) {
+        case 'EVENT_COMMUNICATION':
+            return <i class="fas fa-comment fa-2x" style={{color: isread}}></i>;
+        case 'EVENT_UPDATE':
+            return <i class="fas fa-calendar-day fa-2x" style={{color: isread}}></i>;   
+        case 'PEOPLE_MGMT':
+            return <i class="fas fa-users fa-2x" style={{color: isread}}></i>;
+        default:
+            return "";
+    }
+}
+
+class Notification extends Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            allRead: false,
+            notifications: []
+        };
+        this.isAllRead = this.isAllRead.bind(this);
+        this.isRead = this.isRead.bind(this);
+    }
+
+    componentDidMount() {
+        this.getNotifications();
+    }
+
+    isAllRead(){
+        let read = this.state.notifications.slice(0, 10).filter(e => e.is_read == false).length > 0 ? false : true;
+        if ( read != this.state.allRead){
+            this.setState({
+                allRead: !this.state.allRead
+              });
```

Comment:
Why not just do 
````
this.setState({allRead : read}) 
````
Pree sure that's the same thing? Bc you're saying if they're different, flip allRead. Would be cleaner.

Reply:
Read some documentation, `this.setState` would not work for some reason if it is not wrapped in a conditional. I just tried it again and that was the ase.

Reply:
Hm interesting i'll take a look on it. idk if @zhengwin has ever encountered an issue like that before?

Reply:
Picking up on the comment you left 2 days ago, I thought Jackson discussed the setState issue with you (may have been a few weeks ago)? 

We're calling the function in `render()` rather than a lifecycle like `componentDidMount()` so thats why the traditional setState doesn't work. You're constantly setting state and re-rendering (even if the state being is the same). If you move it out it will work and be cleaner iirc the discussion correctly.